tests/test_connection.py

```diff
@@ -1266,7 +1266,7 @@ def test_no_fee_estimate(node_factory, bitcoind, executor):
 
     # Can do mutual close.
     l1.rpc.close(l2.info['id'])
-    bitcoind.generate_block(100)
+    bitcoind.wait_for_log(r'txs: 1', None, lambda: bitcoind.generate_block(1))
```

Comment:
there may be bitcoind.wait_for_log(r'txs: 1', None, lambda: bitcoind.generate_block(100)) ?

Reply:
As I understand it the only reason it was `generate_block(100)` before is that it was hedging that the single transaction it is interested in is present in one of those 100 blocks.

This is a more precise and more reliable wait.

Reply:
As I understand it the only reason it was `generate_block(100)` before is that it was hedging that the single transaction it is interested in is present in one of those 100 blocks.

This is a more precise and more reliable wait.

Reply:
Nope, this was intentional as we just need one block with the TX. the 100 was overkill to make an indirect 'sleep' effect where eventually the TX got included and mined.